### ISSUE TITLE
[Bug?]: Crashes with TypeError: Cannot read properties of null (reading 'createServerReference')

### ISSUE THREAD

**Issue by bartoszkrawczyk2** [solidjs/solid-start#1223]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Once in a while dev server crashes with an error:
```
TypeError: Cannot read properties of null (reading 'createServerReference')
```

After making some changes and hot reload it disappears.

### Steps to reproduce 🕹

Steps:

1. Create init project
2. Add server cache in Home:
```
const getHello = cache(() => {
  "use server";
  return { message: 'hello; };
}, "hello");

export const route = {
  load: () => getHello(),
};

export default function Home() {
  const hello = createAsync(getHello);
  // ...
```
3. Do some changes in `Home` component
4. Sometimes it crashes, sometimes it resumes. Hard to recreate exact steps, it seems random (I suspect some kind of race condition).

### Your environment 🌎

```shell
System:
    macOS 13.0     M2 Mac

Node:
   node v20.10.0
   npm  v10.2.3
```

**Comment by senabi** [comment]
It crashes for server functions. if you remove the load function, first load doesn't get any errors but if you navigate back, it gets the null error.  The error shows in local production build and dev


```json
  "dependencies": {
    "@solidjs/meta": "^0.29.2",
    "@solidjs/router": "^0.11.5",
    "@solidjs/start": "^0.5.3",
    "solid-js": "^1.8.14",
    "vinxi": "^0.2.1"
  },
```

**Comment by ryansolid** [comment]
I am having no luck reproducing this. I know it is supposed to be hard to reproduce, but I understand the original issue is related to dev environment. I'm just not seeing this no matter what I do.

@senabi are you saying you see this message a different way. Most specifically around `createServerReference`? `Cannot read properties of null` in general can happen for a variety of reasons but I'm interested around `createServerReference`.

**Comment by senabi** [comment]
@ryansolid  sorry for the late response. The exact error I got is `Cannot read properties of null (reading 'startsWith')` using. the basic template. But it seems it's been partially solved by #1328 .

**Comment by gabrielmfern** [comment]
> @ryansolid sorry for the late response. The exact error I got is `Cannot read properties of null (reading 'startsWith')` using. the basic template. But it seems it's been partially solved by #1328 .

I think this was fixed, I also had this. This was fixed by #1329

**Comment by nksaraf** [comment]
Seems like this is fixed. if it's still an issue, reopen this issue. Try with latest solid-start and vinxi.
